openflow13/multipart_test.go

```diff
@@ -0,0 +1,142 @@
+package openflow13
+
+import (
+	"bytes"
+	"testing"
+
+	"antrea.io/libOpenflow/common"
+)
+
+func TestMultipartMessage(t *testing.T) {
+	feature := newTableFeatures()
+	mpartRequest := &MultipartRequest{
+		Header: NewOfp13Header(),
+		Type:   MultipartType_TableFeatures,
+		Flags:  0,
+		Body:   feature,
+	}
+	reqBytes, err := mpartRequest.MarshalBinary()
+	if err != nil {
+		t.Fatalf("function MarshalBinary failed: %v", err)
+	}
+	deReq := new(MultipartRequest)
+	err = deReq.UnmarshalBinary(reqBytes)
+	if err != nil {
+		t.Fatalf("function UnmarshalBinary failed: %v", err)
+	}
+	if !mpartRequestEquals(mpartRequest, deReq) {
+		t.Errorf("Original MultipartRequest not equal to the decoded object")
+	}
+}
+
+func mpartRequestEquals(oriReq, deReq *MultipartRequest) bool {
+	if !headerEquals(oriReq.Header, deReq.Header) {
+		return false
+	}
+	if oriReq.Type != deReq.Type {
+		return false
+	}
+	if oriReq.Flags != deReq.Flags {
+		return false
+	}
+	if oriReq.Body != nil && deReq.Body == nil || oriReq.Body == nil && deReq.Body != nil {
+		return false
+	}
+	if oriReq.Body != nil {
+		switch oriReq.Type {
+		case MultipartType_TableFeatures:
+			if !ofPTableFeaturesEquals(oriReq.Body.(*OFPTableFeatures), deReq.Body.(*OFPTableFeatures)) {
+				return false
+			}
+			break
```

Comment:
not needed?

not sure why we use a `switch` statement to begin with, instead of a simple `if` :)